src/Framework/PaymentGateways/DonationSummary.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Give\Framework\PaymentGateways;
+
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+
+/**
+ * @unreleased
+ */
+class DonationSummary
+{
+    /** @var int */
+    protected $length = 255;
+
+    /** @var GatewayPaymentData */
+    protected $paymentData;
+
+    /**
+     * @unreleased
+     * @param GatewayPaymentData $paymentData
+     */
+    public function __construct( GatewayPaymentData $paymentData )
+    {
+        $this->paymentData = $paymentData;
+    }
+
+    /**
+     * @unreleased
+     * @param $length
+     */
+    public function setLength( $length )
+    {
+        $this->length = $length;
+    }
+
+    /**
+     * @unreleased
+     * @return string
+     */
+    public function getSummaryWithDonor()
+    {
+        return $this->trimAndFilter(implode(' - ', [
+            $this->getSummary(),
+            $this->getDonorLabel(),
+        ]));
+    }
+
+    /**
+     * @unreleased
+     * @return string
+     */
+    public function getSummary()
```

Comment:
@kjohnson are we following the same template for donation summary given in `give_payment_gateway_donation_summary` function?

Reply:
Yes, I test this against the `give_payment_gateway_donation_summary` function directly to ensure the same return value.

```php
$this->assertEquals(
    give_payment_gateway_donation_summary($this->get_legacy_donation_data($paymentData), false),
    $summary->getSummary()
);
```